src/Microsoft.DocAsCode.Utility/UriUtility.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Utility
+{
+    using System;
+
+    public static class UriUtility
+    {
+        private const string QueryMarker = "?";
+        private const string FragmentMarker = "#";
+
+        public static bool HasFragment(string uriString)
+        {
+            if (string.IsNullOrEmpty(uriString)) throw new ArgumentNullException(nameof(uriString));
+            return uriString.IndexOf(FragmentMarker) != -1;
+        }
+
+        public static string GetFragment(string uriString)
+        {
+            if (string.IsNullOrEmpty(uriString)) throw new ArgumentNullException(nameof(uriString));
+            var index = uriString.IndexOf(FragmentMarker);
```

Comment:
Xref still not support query string for now. I'm consider add some other methods that also consider queryMarker later.

---
In reply to: [86513116](https://github.com/dotnet/docfx/pull/860#discussion_r86513116) [](ancestors = 86513116)